### PR TITLE
Abstract Navigation Logic Into External Package

### PR DESCRIPTION
**Pull request by abeldemoz**

### CODE REVIEW

File: Under The Weather/Navigation/Coordinators/WeatherCoordinator.swift [hunk 1]
```diff
@@ -5,64 +5,58 @@
...
+    func citySelectionNextTapped() {
+        navigator.dismiss(animated: true)
+    }
 
-    typealias Factory = WeatherViewControllerFactory
-    & CitySearchViewControllerFactory
-    & AboutViewControllerFactory
+    typealias Factory = (WeatherViewControllerFactory
+        & CitySearchCoordinatorFactory
+        & AboutViewControllerFactory)
```
Comment (mattybwoy) [inline Under The Weather/Navigation/Coordinators/WeatherCoordinator.swift]: Not sure why there's brackets here

File: Under The Weather/Networking/NetworkService.swift [hunk 2]
```diff
@@ -8,65 +8,62 @@
...
-            
+
+        let task = urlSession.dataTask(with: WeatherEndpoint.citySearch(with: city).url) { data, _, error in
+
             guard let data = data, error == nil else {
                 completionHandler(.failure(NetworkError.invalidSearch))
                 return
             }
             do {
                 let response = try
```
Comment (mattybwoy) [inline Under The Weather/Networking/NetworkService.swift:35]: Isn't response captured from above, but you've replaced it with _

File: Under The Weather/SceneDelegate.swift [hunk 3]
```diff
@@ -20,7 +21,8 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         let dependencyContainer = DependencyContainer()
         let navigator = AppNavigator(window: window!)
         let coordinator = AppCoordinator(navigator: navigator, factory: dependencyContainer)
-        coordinator.start(animated: true, onDismissed: nil)
+        let navigationController = BasicNavigationController()
```
Comment (mattybwoy) [inline Under The Weather/SceneDelegate.swift]: navigationController is instantiated here but we don't use it?

**Review by mattybwoy — COMMENTED** [review]
Finished looking over this juicy one, just couple of comments

File: Under The Weather/Navigation/Coordinators/WeatherCoordinator.swift (hunk 1 above)
Comment (abeldemoz) [reply Under The Weather/Navigation/Coordinators/WeatherCoordinator.swift]: Parentheses make no difference here, they're not really needed

File: Under The Weather/Networking/NetworkService.swift (hunk 2 above)
Comment (abeldemoz) [reply Under The Weather/Networking/NetworkService.swift:35]: `response` isn't used within the closure

File: Under The Weather/SceneDelegate.swift (hunk 3 above)
Comment (abeldemoz) [reply Under The Weather/SceneDelegate.swift]: Either the linter or compiler should have caught that. Good catch, I'll remove it in the next commit 👍🏽

**Pull request merged**